cpp/oneapi/dal/backend/interop/table_conversion.hpp

```diff
@@ -46,19 +36,34 @@ inline auto allocate_daal_homogen_table(std::int64_t row_count, std::int64_t col
         daal::data_management::NumericTable::doAllocate);
 }
 
-template <typename T>
-inline auto convert_to_daal_homogen_table(array<T>& data,
+template <typename Data>
+inline auto convert_to_daal_homogen_table(array<Data>& data,
                                           std::int64_t row_count,
                                           std::int64_t column_count) {
     if (!data.get_count())
-        return daal::services::SharedPtr<daal::data_management::HomogenNumericTable<T>>();
+        return daal::services::SharedPtr<daal::data_management::HomogenNumericTable<Data>>();
     data.need_mutable_data();
     const auto daal_data =
-        daal::services::SharedPtr<T>(data.get_mutable_data(), daal_array_owner<T>{ data });
+        daal::services::SharedPtr<Data>(data.get_mutable_data(), daal_array_owner{ data });
+
+    return daal::data_management::HomogenNumericTable<Data>::create(daal_data,
+                                                                    column_count,
+                                                                    row_count);
+}
+
+template <typename AlgorithmFPType>
```

Comment:
Arguable, because I want to mark that this type came from the algorithm - it is the main type for algorithm computations. 
It is used as an auxiliary type for converting a table. The main goal of the functions of table conversion is not to change its internal data type - if the input is homogen table on `double` but `AlgorithmFpType` is `float` it is better to wrap homogen table with the old one with `double` type. If we cannot save original data types/layout, then `AlgorithmFpType` is used

Reply:
Can this function be invoke with `int64_t` or `int32_t`?  Moreover, `AlgorithmFPType`  is not used in new interfaces elsewhere, instead we use `Float` to refer to `float` or `double` incl. kernels for algorithms.

Reply:
> Can this function be invoke with int64_t or int32_t
No, since `train()` and `infer()` can be invoked just with `float` or `double` types

Reply:
renamed to `Float`